### PR TITLE
Beta.1 - Adding new database features

### PR DESCRIPTION
**Pull request by joscdk**
Adding the new database features for the beta.

- [x] Create new database server
- [x] Create database for each environment
- [x] Database login
- [x] Database show info
- [x] Resize database server
- [x] Restart database server
- [x] Shutdown database server
- [x] Delete database server
- [x] Implement token to support multiple database servers
- [x] Add note when people need to redeploy
- [x] Automatically add/modify configuration variables

**Backup**
TBD

### CODE REVIEW

File: Sources/Cloud/Commands/Database/DatabaseCreate.swift [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        
+        console.pushEphemeral()
+        
+        size = try console.giveChoice(
+            title: "Which size?",
+            in: Size.all
+        ) { type in
+            switch type {
+            case .free:
+                return "(Dev) Free $0/month - (256mb memory - 10,000 records - 20 connections - no backups)"
```
Comment (tanner0101) [inline Sources/Cloud/Commands/Database/DatabaseCreate.swift]: 1: what does `(Dev)` mean?
2: slashes could be nice `(256MB memory / 10K Rows / 20 Connections)`
we probably don't need to say "no backups", just saying "yes backups" for the plans that have them would work

File: Sources/Cloud/Commands/Database/DatabaseCreate.swift [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        console.info("This creates a database server for the application: \(app)")
+        console.info("It will also setup a database on the server for each of the environments:")
+        
+        try environments.forEach { val in
+            console.info(" - \(val)")
+            envArray.append("\(val.id ?? "")")
+        }
+        console.info("")
+        
+        let name = console.ask("What name for this database server? [a-zA-Z0-9]")
```
Comment (tanner0101) [inline Sources/Cloud/Commands/Database/DatabaseCreate.swift]: we should require uppercase. `[A-Z0-9]`

File: Sources/Cloud/Commands/Database/DatabaseFeedbackFormat.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            do {
+                exit(0)
+            } catch {
+                exit(1)
+            }
+        }
+        
+        switch(info.status) {
+        case "error":
+            console.error("[ Error ] " + info.message)
```
Comment (tanner0101) [inline Sources/Cloud/Commands/Database/DatabaseFeedbackFormat.swift]: hmm, why is this a different format than the other errors? can we use the loading bar here?

File: Sources/Cloud/Commands/Database/DatabaseInspect.swift [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    ]
+    
+    public let signature: [Argument] = [
+        Option(name: "app", help: [
+            "The slug name of the application to deploy",
+            "This will be automatically detected if your are",
+            "in a Git controlled folder with a remote matching",
+            "the application's hosting Git URL."
+            ]),
+        Option(name: "token", help: [
```
Comment (tanner0101) [inline Sources/Cloud/Commands/Database/DatabaseInspect.swift]: might be nice to have this static and shared.

File: Sources/Cloud/Commands/Database/DatabaseFeedbackFormat.swift (hunk 3 above)
Comment (joscdk) [reply Sources/Cloud/Commands/Database/DatabaseFeedbackFormat.swift]: Loading bars will be implemented, but requires a lot of additional work to make it work properly :) .. Again, this is a very early beta, to get the database part of things tested, that will take months to get tested properly, so just want to get it out, so we can start shooting down bugs in the actual databases, sugar can always be added later :)

File: Sources/Cloud/Commands/Database/DatabaseCreate.swift (hunk 1 above)
Comment (joscdk) [reply Sources/Cloud/Commands/Database/DatabaseCreate.swift]: Dev relates to the different plans, dev meaning free/hobby .. Later they will be changed to questions, so you select:

- Dev
- Standard
- Professional

And then the specific plan after :) .. Just to stress out now it's not for production in any way :)

**Pull request merged**